src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java

```diff
@@ -27,6 +27,20 @@ public SAMRecordToGATKReadAdapter( final SAMRecord samRecord ) {
         this.samRecord = samRecord;
     }
 
+    /**
+     * Produces a SAMRecordToGATKReadAdapter wrapping the provided SAMRecord,
+     * and nulls out the header in the encapsulated read. This is useful for
+     * Spark tools, in order to avoid serializing the SAMFileHeader for each
+     * record.
+     *
+     * @param samRecord read to adapt (header will be stripped)
+     * @return SAMRecordToGATKReadAdapter wrapping the headerless samRecord
+     */
+    public static SAMRecordToGATKReadAdapter headerlessReadAdapter( final SAMRecord samRecord ) {
```

Comment:
Because non-spark tools don't need to operate on reads with null headers (and take the risks associated with doing so)